Build failed on Windows

**What component of `google-cloud-cpp` is this related to?**
`google/cloud/storage`

**Describe the bug**
Build failed on Windows. These lines below cause `error C2593: 'operator =' is ambiguous`. 
https://github.com/googleapis/google-cloud-cpp/blob/87d600391bb245fe7734ded9d032de595bd21f8d/google/cloud/storage/parallel_upload.cc#L132
https://github.com/googleapis/google-cloud-cpp/blob/87d600391bb245fe7734ded9d032de595bd21f8d/google/cloud/storage/parallel_upload.cc#L151
Maybe we should cast explicitly `destinasion_json` and `custom_data_json` to `std::string`
I could make a PR to this issue if you need.
```
res.destination_object_name = destination_json.get<std::string>()
```

**To Reproduce**
Build on windows

**Expected behavior**
Successful build

**Operating system:**
Windows 10

**What compiler and version are you using?**
Microsoft (R) C/C++ Optimizing Compiler Version 19.25.28612 for x64

**What version of `google-cloud-cpp` are you using?**
https://github.com/googleapis/google-cloud-cpp/releases/tag/v1.13.0

--- a/google/cloud/storage/parallel_upload.cc
+++ b/google/cloud/storage/parallel_upload.cc
@@ -129,7 +129,7 @@ ParallelUploadPersistentState::FromString(std::string const& json_rep) {
     return Status(StatusCode::kInternal,
                   "Parallel upload state's 'destination' is not a string.");
   }
-  res.destination_object_name = destination_json;
+  res.destination_object_name = destination_json.get<std::string>();
   if (json.count("expected_generation") != 1) {
     return Status(
         StatusCode::kInternal,
@@ -148,7 +148,7 @@ ParallelUploadPersistentState::FromString(std::string const& json_rep) {
       return Status(StatusCode::kInternal,
                     "Parallel upload state's 'custom_data' is not a string.");
     }
-    res.custom_data = custom_data_json;
+    res.custom_data = custom_data_json.get<std::string>();
   }
   if (json.count("streams") != 1) {
     return Status(StatusCode::kInternal,

